### PR TITLE
Blueprint toaster triggering warning fix

### PR DESCRIPTION
**Pull request by Ryanseanlee**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [ ] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [ ] Updated doc comments / prop-types, or determined not required.
- [ ] Reviewed and tested on Mobile, or determined not required.
- [ ] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: desktop/appcontainer/ToastSource.ts [hunk 1]
```diff
@@ -80,7 +82,7 @@ class ToastSourceLocalModel extends HoistModel {
      * @param position - position on screen where toast should appear.
      * @param containerRef - DOM Element used to position (contain) the toast.
      */
-    getToaster(position: ToasterPosition, containerRef: HTMLElement) {
+    async getToaster(position: ToasterPosition, containerRef: HTMLElement) {
         if (containerRef && !isElement(containerRef)) {
```
Comment (lbwexler) [inline desktop/appcontainer/ToastSource.ts:86]: name should have async suffix -- this is an internal API, so ok to change that

File: desktop/appcontainer/ToastSource.ts [hunk 2]
```diff
@@ -94,8 +96,36 @@ class ToastSourceLocalModel extends HoistModel {
...
-            toasters[position] = OverlayToaster.create({position, className}, container);
+            toasters[position] = await this.createToaster({position, className}, container);
         toasterMap.set(container, toasters);
         return toasters[position];
     }
+
+    /** Workaround to avoid calling OverlayToaster.create(). It uses ReactDOM.render
+     * that gives a warning because it is deprecated in React 18.
+
+     * Is currently set to be removed in Blueprint v6.0
```
Comment (lbwexler) [inline desktop/appcontainer/ToastSource.ts:107]: what is set to be removed?  The create method?  Is this code from Blueprint v6?

Comment (lbwexler) [reply desktop/appcontainer/ToastSource.ts:107]: Unless we are explicitly trying not to touch this new `createToaster` code,  the signature can be simplified -- it always takes props, and an element, and returns a Promise<OverlayToaster>.

File: desktop/appcontainer/ToastSource.ts [hunk 3]
```diff
@@ -94,8 +96,36 @@ class ToastSourceLocalModel extends HoistModel {
...
+        const containerElement = document.createElement('div');
+        container.appendChild(containerElement);
+        const root = createRoot(containerElement);
+        return new Promise<OverlayToaster>((resolve, reject) => {
+            root.render(
+                overlayToaster({
+                    ...props,
+                    usePortal: false,
+                    ref: instance => {
+                        if (!instance) {
```
Comment (lbwexler) [inline desktop/appcontainer/ToastSource.ts:119]: reverse the sense of the if  for clarity

File: desktop/appcontainer/ToastSource.ts (hunk 2 above)
Comment (Ryanseanlee) [reply desktop/appcontainer/ToastSource.ts:107]: So react deprecated the ReactDOM.render and current blueprint v5 toaster.create uses ReactDOM.render.(the reason we are getting the warning) They have notes that say they plan to remove the use of ReactDOM.render in v6.
https://github.com/palantir/blueprint/issues/5212#issuecomment-1294958195

**Comment by lbwexler** [comment]
got it -- this link in a code comment would have been all I needed --
whenever we are copying workarounds a link like this is golden.

**Pull request merged**